coala_quickstart/green_mode/green_mode.py

```diff
@@ -0,0 +1,537 @@
+    for index, obj in enumerate(contents[settings_key]):
+        if isinstance(obj, dict) and key in obj.keys():
+            found = True
+            contents[settings_key][index][key] += values
+    if not found:
+        contents[settings_key].append({key: values})
+
+    return contents
+
+
+def dump_to_file(file, contents):
+    """
+    Writes YAML data to a file.
+    :param file:
+        The file to write YAML data to.
+    :param contents:
+        The python objects to be written as YAML data.
+    """
+    with open(file, 'w+') as outfile:
+        yaml.dump(contents, outfile,
+                  default_flow_style=False)
+
+
+def initialize_project_data(dir, ignore_globs):
+    """
+    Generates the values for the key 'dir_structure'
+    for `.project_data.yaml` which is directories as
+    dicts with key name as dir names and value as
+    list of files or nested dict of dirs.
+    :param dir:
+        The directory for which to add the directory structure
+        and files.
+    :param ignore_globs:
+        The globs of files to ignore from writing to the
+        '.project_data.yaml'.
+    :return:
+        The python object that was written as YAML data
+        to `.project_data.yaml`.
+    """
+    files_dirs = os.listdir(dir)
+    # files_dirs holds names of both files and dirs.
+    dir_name = dir[dir.rfind(os.sep)+1:]
+    final_data = []
+
+    for i in files_dirs:
+        to_continue = False
+        for glob in ignore_globs:
+            if fnmatch.fnmatch(dir+i, glob):
+                to_continue = True
+        if to_continue is True:
+            continue
+        if os.path.isfile(dir+i):
+            final_data.append(i)
+        else:
+            look_into_dir = dir+i+os.sep
+            data = initialize_project_data(look_into_dir,
+                                           ignore_globs)
+            final_data.append({i: data})
+    return final_data
+
+
+def generate_complete_filename_list(contents, project_dir):
+    """
+    Generates only a list of files with complete file
+    paths from the 'dir_structure' key of `.project_data.yaml`.
+    :param contents:
+        The python object which contains the already parsed
+        directories and files.
+    :param project_dir:
+        The current directory from which to generate the list
+        of files and directories within itself.
+    :return:
+        List of files or nested lists of directories and the
+        structures within it to be appended to the contents
+        from where the function is called.
+    """
+    prefix = project_dir + os.sep
+    file_names_list = []
+    for item in contents:
+        if not isinstance(item, dict):
+            file_names_list.append(prefix + item)
+        else:
+            file_names_list += generate_complete_filename_list(
+                item[next(iter(item))], prefix+next(iter(item)))
+    return file_names_list
+
+
+def find_max_of_setting(setting, value, contents):
```

Comment:
new module `green_mode/Settings.py` for this and other functions which operate on Setting objects .. ?

Reply:
too many files would be named `Settings.py`. Some other name maybe?

Reply:
Nope.

The one in `generation/` is clearly different in purpose to this one.